benchmarks/linalg_refactor_benchmark.cpp

```diff
@@ -0,0 +1,94 @@
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalgrefactor/linalgRefactor.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/linalg/linalg.h>
+#include <vector>
+#include <algorithm>
+#include <memory>
+#include <hayai/hayai.hpp>
+
+#include <iostream>
+
+using namespace shogun;
+
+/**
+ * Instructions :
+ * 1. Install benchmarking toolkit "hayai" (https://github.com/nickbruun/hayai)
+ * 2. Compile against libhayai_main, e.g.
+ * 		g++ -O3 -std=c++11 linglg_refactor_benchmark.cpp -I/usr/include/eigen3 \
+ *		-lshogun -lhayai_main -lOpenCL -o benchmark
+ * 3. ./benchmark
+ */
+
+/** Generate data only once */
+struct Data
+{
+	Data()
+	{
+		A = init(1);
+		B = init(2);
+		Ac = init_c(A);
+		Bc = init_c(B);
+        Ag = init_g(A);//GPU_Vector<int32_t>(A);
+        Bg = GPU_Vector<int32_t>(B);
+	}
+
+	SGVector<int32_t> init(int32_t begin)
+	{
+
+        index_t n = num_rows;
+        std::vector<int32_t> mem(n);
```

Comment:
I'll checkout. thx!
BTW there's a similar step in max_product_benchmark, but there're some bugs..It copies an empty vector to SGMatrix and returned a deleted SGMatrix. Not sure whether you want to fix it or I should.